### DISCUSSION TITLE
ICU4C on Android

### DISCUSSION THREAD

**Discussion by ProgrammerBassam** [florisboard/florisboard#1920] [Q&A]
Hi

According to google
[https://developer.android.com/guide/topics/resources/internationalization#icu4c](url) 

"Android exposes a subset of the ICU4C APIs via the libicu.so library, rather than libicuuc.so or libicui18n.so. The APIs are available starting with Android 12 (API level 31). The NDK headers are available starting with the NDK release r22b. No C++ API is exposed through the Android NDK. Some of the C APIs are not available yet."

Is it match with floris?
In Floris code library is "libicuuc.so" not "libicu.so"

**Comment by patrickgold** [comment]
> The APIs are available starting with Android 12 (API level 31)

Working around API restrictions when it comes to native library linking is hard and it would require me to maintain two APKs, one for Android 7-11 and one for Android 12+, so I am not considering using this. However I will most likely rewrite the spell checker and with that remove the requirement for native ICU4C altogether, so this may not be relevant at all anymore in a few months.
